content/en/docs/languages/java/configuration.md

```diff
@@ -1,185 +1,297 @@
 ---
-title: Configuration
-linkTitle: Configuration
-weight: 10
+title: Configure the SDK
+linkTitle: Configure the SDK
+weight: 12
 aliases: [config]
 # prettier-ignore
-cSpell:ignore: authservice blrp Dotel ignore LOWMEMORY myservice ottrace PKCS retryable tracepropagators
+cSpell:ignore: authservice blrp Dotel ignore LOWMEMORY myservice ottrace PKCS retryable tracepropagators autoconfigured Customizer
 ---
 
-The OpenTelemetry SDK provides a working implementation of the API, and can be
-set up and configured in a number of ways. The Java SDK supports most of the
-available [configuration options](/docs/languages/sdk-configuration/). For
-conformance details, see the
-[compliance matrix](https://github.com/open-telemetry/opentelemetry-specification/blob/main/spec-compliance-matrix.md).
+<!-- markdownlint-disable blanks-around-fences -->
+<?code-excerpt path-base="examples/java/configuration"?>
+
+The [SDK](/docs/languages/java/sdk/) is the built-in reference implementation of
+the [API](/docs/languages/java/instrumentation/), processing and exporting
+telemetry produced by instrumentation API calls. Configuring the SDK to process
+and export appropriately is an essential step to integrating OpenTelemetry into
+an application.
+
+All SDK components have
+[programmatic configuration APIs](#programmatic-configuration). This is the most
+flexible, expressive way to configure the SDK. However, changing configuration
+requires adjusting code and recompiling the application, and there is no
+language interoperability since the API is written in java.
+
+The [zero-code SDK autoconfigure](#zero-code-sdk-autoconfigure) module offers a
+compelling alternative, configuring SDK components based off system properties /
+environment variables, with various extension points for instances where the
+properties are insufficient. **We recommend the zero-code SDK autoconfigure
+module.**
```

Comment:
```suggestion
The [zero-code SDK autoconfigure](#zero-code-sdk-autoconfigure) module 
configures SDK components through system properties or environment
variables, with various extension points for instances where the properties
are insufficient.
```

Consider moving the recommendation to a callout in imperative tense that explains why using zero-code is a better choice.